python/demos/picture of the day viewer/app.py

```diff
@@ -0,0 +1,160 @@
+    MIT License
+"""
+
+#!/usr/bin/python3
+
+from datetime import date, timedelta
+from flask import Flask, render_template, request
+import requests
+
+APP = Flask(__name__)
+SESSION = requests.Session()
+ENDPOINT = "https://en.wikipedia.org/w/api.php"
+_current_date = date.today()
+
+@APP.route("/", methods=["GET", "POST"])
+
+def index():
+    """
+    Requests data from the WikiMedia Action API, related to the POTD for
+    the _current_date, and renders it on a webpage located at the index path.
+
+    Returns
+    -------
+    JSON data related to the current POTD, if POST request.
+    Renders the webpage, templates/index.html, otherwise.
+    """
+
+    global _current_date
+
+    if request.method == "POST":
+        _current_date = change(_current_date)
+
+    data = fetch_potd(_current_date)
+
+    return render_template("index.html", data=data)
+
+
+def change(date_object):
+    """
+    Return new date in response to user input, thus changing the POTD
+    being displayed on the webpage.
+    """
+
+    user_input = request.form["change_date"]
+    new_date = _current_date
+    last_date = date.today()
+    first_date = date(year=2004, month=5, day=14)
+
+    if user_input == "← Back" and decrement(new_date) > first_date:
+        new_date = decrement(new_date)
+    elif user_input == "Next →" and increment(new_date) <= last_date:
+        new_date = increment(new_date)
+
+    return new_date
+
+def increment(date_object):
+    """Returns the next date for a date object (i.e. tomorrow)"""
+
+    return date_object + timedelta(days=1)
+
+def decrement(date_object):
+    """Returns the previous date for a date object (i.e. yesterday)"""
+
+    return date_object - timedelta(days=1)
+
+
+def fetch_potd(date_object):
+    """
+    Returns image data related to the current POTD.
+
+    Parameters
+    ----------
+    date_object: date
+        A datetime.date object, used to construct a title in order to find the
+        correct POTD in the POTD archives.
+
+    Returns
+    -------
+    list[dict]
+        JSON detailing the POTD's filename, image url, description url, and the
+        date it was featured as Picture of the Day.
+    """
+
+    date_string = date_object.isoformat()
+
+    parameters = {
+        "action": "query",
+        "format": "json",
+        "formatversion": "2",
+        "prop": "images",
+        "titles": "Template:POTD protected/" + date_string
+    }
+
+    response = SESSION.get(url=ENDPOINT, params=parameters)
+    data = response.json()
+
+    file_name = data["query"]["pages"][0]["images"][0]["title"]
+    image_info = fetch_image_info(file_name)
+    potd_page = "https://en.wikipedia.org/wiki/Template:POTD_protected/" + date_string
+
+    results = {
+        "title": file_name,
+        "image_src": image_info,
+        "image_page": potd_page,
+        "date": date_object
+    }
+
+    return results
+
+def fetch_image_info(file_name):
+    """
+    Returns the POTD's image url & image description.
```

Comment:
We are not sending image URL, so may be just say that in the description only. Also, may be we need to change the function name to reflect the same.

Reply:
I updated this to remove references to image description. I think that's what you meant